src/simulate.c

```diff
@@ -99,6 +97,35 @@ void simulate(
     sim_data sim;
     sim_init(&sim, sim_offload);
 
+#ifdef GPU
+    if(sim.sim_mode != 1) {
+        print_err("Only GO mode ported to GPU. Please set SIM_MODE=1.");
+        exit(1);
+    }
+    if(sim.record_mode) {
+        print_err("RECORD_MODE=1 not ported to GPU. Please disable it.");
+        exit(1);
+    }
+    if(sim.enable_atomic) {
+        print_err("Atomic not yet ported to GPU. Please set ENABLE_ATOMIC=0.");
+        exit(1);
+    }
+    if(sim.enable_mhd) {
+        print_err("MHD not yet ported to GPU. Please set ENABLE_MHD=0.");
+        exit(1);
+    }
+    if(sim.diag_data.diagorb_collect) {
+        print_err(
+            "ENABLE_ORBITWRITE=1 not ported to GPU. Please disable it.");
+        exit(1);
+    }
+    if(sim.diag_data.diagtrcof_collect) {
+        print_err(
+            "ENABLE_TRANSCOEF=1 not ported to GPU. Please disable it.");
+        exit(1);
+    }
```

Comment:
Here, I'm not sure that `sim.diag_data.diagtrcof_collect` is filled. Indeed, `diag_init` is called after in line 176. And in `diag_init` (line 154 in `diag.c`  `sim.diag_data.diagtrcof_collect` is filled `  data->diagtrcof_collect = offload_data->diagtrcof_collect;`
Maybe put after these tests testing if GPU is enabled ?

Reply:
I replaced the sim with the sim_offload struct which is fully initialized. Now it should work!